types/polymer/index.d.ts

```diff
@@ -163,7 +163,7 @@ declare global {
 
       getContentChildren?(selector: string): HTMLElement[];
 
-      fire?(type: string, detail?: any, options?: Object): CustomEvent;
+      fire?<Detail>(type: string, detail?: Detail, options?: Object): CustomEvent;
```

Comment:
I'd like to avoid adding a type parameter to `fire`, as the long term goal is to be able to read the tagname from the class and use that to look up event type information.

Something like:

```typescript
  fire<K extends keyof EventDetailTagnameMap[this.constructor.is]>(
    type: K, detail?: EventDetailTagnameMap[this.constructor.is][K], options?: {bubbles: boolean}): CustomEvent;
```

Adding manual annotation to `fire` will make it a breaking change to move to this tagname based system.